users/src/main/java/io/crate/operation/auth/TrustAuthentication.java

```diff
@@ -29,12 +30,16 @@
 public class TrustAuthentication implements AuthenticationMethod {
 
     static final String NAME = "trust";
-    private static final String SUPERUSER = "crate";
+    private final UserManager userManager;
+
+    public TrustAuthentication(UserManager userManager) {
```

Comment:
But a `Predicate<String>` would work. The ctor would then just be called as `new TrustAuthentication(userManager::userExists)`